core/controllers/acl_decorators.py

```diff
@@ -2318,11 +2338,47 @@ def test_can_access(self, topic_name, **kwargs):
         Raises:
             PageNotFoundException: The given page cannot be found.
         """
-        topic_name = python_utils.url_unquote_plus(topic_name)
-        topic = topic_fetchers.get_topic_by_name(topic_name)
+        lowercase_abbrev_topic_name = (
+            python_utils.url_unquote_plus(abbrev_topic_name).replace('-', ' '))
```

Comment:
Why do you need url_unquote_plus()? There is no indication that abbreviated_topic_name is part of a URL, at least from the variable name.

Also, why don't you just have the "abbreviated name" in the topic store the URL fragment directly, rather than doing all this encoding? It's more direct. You can also have that field include validation to ensure that only a very restricted subset of characters is used.